backend/core/src/main/java/org/sonarsource/sonarlint/core/issue/IssueService.java

```diff
@@ -450,18 +459,22 @@ private static RaisedIssueDto updateIssue(RaisedIssueDto issue, Set<String> impa
 
   private void updateProjectIssueStorage(String connectionId, IssueChangedEvent event) {
     var findingsStorage = storageService.connection(connectionId).project(event.getProjectKey()).findings();
-    event.getImpactedIssueKeys().forEach(issueKey -> findingsStorage.updateIssue(issueKey, issue -> {
+    event.getImpactedIssues().forEach(issue -> findingsStorage.updateIssue(issue.getIssueKey(), storedIssue -> {
       var userSeverity = event.getUserSeverity();
       if (userSeverity != null) {
-        issue.setUserSeverity(userSeverity);
+        storedIssue.setUserSeverity(userSeverity);
       }
       var userType = event.getUserType();
       if (userType != null) {
-        issue.setType(userType);
+        storedIssue.setType(userType);
       }
       var resolved = event.getResolved();
       if (resolved != null) {
-        issue.setResolved(resolved);
+        storedIssue.setResolved(resolved);
+      }
+      var impacts = issue.getImpacts();
+      if (!impacts.isEmpty()) {
+        storedIssue.setImpacts(impacts);
```

Comment:
Is `getImpacts` in this event returning the overridden impacts or the full list? This [ticket](https://sonarsource.atlassian.net/browse/SONAR-23287) suggests it's only the "change", not sure what that means. On the other hand, this [ticket](https://sonarsource.atlassian.net/browse/SONAR-23429) mentions that the `issues/pull` endpoint returns the full list of impacts. This means you are calling `setImpacts` with a different set of impacts, but these assumptions need to be verified. Then I would suggest:
* merge impacts from the events with the ones already present in the issue
* add a test to cover this scenario

One last thing: how do you distinguish between no impacts in the event (because something else changed, like the resolution) and overridden impacts being removed? This if might prevent you from clearing the overrides in the storage

Reply:
After confirmation, once a rule has been modified, the impact field will always be present, even when the impact is modified back to the original.